sophia/src/term/iri/_join.rs

```diff
@@ -131,11 +131,11 @@ impl<'a> IriParsed<'a> {
     }
 }
 
-impl<'a> Resolve<String> for IriParsed<'a> {
+impl<'a> Resolve<&str, String> for IriParsed<'a> {
     /// Resolve an IRI given as `String`.
     ///
     /// Fails if `other` is not a valid IRI.
-    fn resolve(&self, other: &String) -> Result<String> {
+    fn resolve(&self, other: &&str) -> Result<String> {
```

Comment:
`&&str` is ackward... Why not just
```rust
impl<'a> Resolve<str, String> for IriParsed<'a> {
    fn resolve(&self, other: &str) -> Result<String> {
       // ...
    }
}
```
?

Reply:
The trait is defined as follows:

```rust
pub trait Resolve<S, T = S> {
    fn resolve(&self, other: &S) -> Result<T>;
}
```

which means that `T` must be `Seized` but `str` isn't so your proposal doesn't work for this definition. But your right it would be nicer with only `str`. I'll give it a try but it'll need some trickery with trait bounds.

Reply:
This issue was resolved in 4b84a9a